src/cls/rbd/cls_rbd_types.h

```diff
@@ -495,6 +508,69 @@ struct TrashSnapshotNamespace {
   }
 };
 
+struct MirrorPrimarySnapshotNamespace {
+  static const SnapshotNamespaceType SNAPSHOT_NAMESPACE_TYPE =
+    SNAPSHOT_NAMESPACE_TYPE_MIRROR_PRIMARY;
+
+  bool demoted = false;
+  std::set<std::string> mirror_peers;
```

Comment:
lgtm

Reply:
@dillaman Actually thinking a little more about this. Using the pool mirror_uuid from `ping` would be possible only if we had two-way mirroring configured. So for one-way mirroring we would still need to look-up the mirror peer ID in the remote cluster?

Reply:
If the remote site isn't pinging the local cluster, it means it doesn't have an `rbd-mirror` daemon running which implies it would be an RX-only peer (so you wouldn't need to add it to the set of peers within the snapshot).

Reply:
Ah, I was initially confused. Now I think I see what you meant. Just to make sure, do you want the ping message update the existing `uuid` field in the `MirrorPeer` or it should be a new field, like `remote_mirror_uuid`? I don't quite remember how currently `uuid` is used but I have a suspicion that updating it might break something. Actually, what it the current purpose of this field?

The disadvantage I think is that the user can't start creating mirroring snapshots until the rbd mirror is first started. Probably not big deal but we need to make sure there is no confusion on the user side.

But also with this approach it seems there is a little difference if we use `uuid` (from ping) or `fsid` (from ping). And as the fsid is already here, then probably I will just add  `mirror_peer_fsids` instead of `mirror_peer_uuids`?

Reply:
I was saying we should just put a new field in `MirrorPeer` for `mirror_uuid` -- the existing `uuid` field is the mirror_peer_uuid duplicated. The `fsid` also works but it would have the same limitation that the peer must have already connected at least once to work.

Reply:
Yes, `fsid` would have the same limitation, and that is why I didn't like it initially. But if the alternative is to add yet another field that would serve similar purposes, then I vote for just using `fsid`. It is already here and it seems it will be less confusing for users. And we could allow to set peer's fsid manually via CLI/UI when configuring the peer to make the snapshots work before the peer is connected (the same would be true for the newly added `mirror_uuid` but it would probably harder for user to understand what uuid is meant).

Reply:
So, I vote either using local mirror peer uuids (as I intended initially and as it is in my current implementation) or fsids. What do you prefer?

Reply:
I can live w/ the `mirror_peer_uuid` -- but let's just name the variables accordingly throughout.